glooshot/docs/tutorial/bookinfo_tutorial.md

```diff
@@ -0,0 +1,265 @@
+```
+
+- Verify that the app is ready.
+  - When the pods in the `default` namespace are ready, we can start testing our app
+```bash
+kubectl get pods -n default -w
+```
+
+- Let's access the landing page of our app
+  - Execute the command below.
+  - Navigate to http://localhost:9080/productpage?u=normal in your browser.
+  - You should see a book description, reviews, and ratings - each provided by their respective services.
+  - Reload the page a few times, notice that the ratings section changes. Sometimes there are no stars, other times red or black stars appear. This is because Istio is load balancing across the four versions of the reviews service. Each reviews service renders the ratings data in a slightly different way.
+```bash
+kubectl port-forward -n default deployment/productpage-v1 9080
+```
+
+- Let's use Supergloo to modify Istio's configuration such that all reviews requests are routed to the version of the service that has red stars - and an **unknown vulnerability!**
+  - Execute the command below
+  - Now when you refresh the page, the stars should always be red.
+```bash
+supergloo apply routingrule trafficshifting \
+    --namespace glooshot \
+    --name reviews-v4 \
+    --dest-upstreams glooshot.default-reviews-9080 \
+    --target-mesh glooshot.istio-istio-system \
+    --destination glooshot.default-reviews-v4-9080:1
+```
+
+
+### Create an experiment
+
+- Create a simple experiment with `kubectl`
+  - We will create a fault on the ratings service such that it always returns `500` as a response code.
+  - We will run this experiment with the following conditions:
+    - The prometheus query `scalar(sum(istio_requests_total{ source_app="productpage",response_code="500"}))` must not exceed a threshold of 10.
+    - The experiment should expire after 600 seconds
+  - Execute the command below to create this experiment
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: glooshot.solo.io/v1
+kind: Experiment
+metadata:
+  name: abort-ratings-metric
+  namespace: default
+spec:
+  spec:
+    duration: 600s
+    failureConditions:
+      - prometheusTrigger:
+          customQuery: |
+            scalar(sum(istio_requests_total{ source_app="productpage",response_code="500"}))
+          thresholdValue: 10
+          comparisonOperator: ">"
+    faults:
+    - destinationServices:
+      - name: default-ratings-9080
+        namespace: glooshot
+      fault:
+        abort:
+          httpStatus: 500
+        percentage: 100
+    targetMesh:
+      name: istio-istio-system
+      namespace: glooshot
+EOF
+```
+
+- Refresh the page, you should now see a failure: none of the reviews data is rendered
+- Refresh the page about 10 more times.
+- Within 15 seconds after the threshold value is exceeded you should see the error go away. The experiment stop condition has been met and the fault that caused this cascading failure has been removed.
```

Comment:
Why is it within 15s after the threshold value? What's special about 15s?

Reply:
Ah -- just realized -- is this a refresh rate that is defined somewhere that would notice the prometheus metric?

Reply:
If so... might be worth helping the reader understand this